packages/math/texlike.lua

```diff
@@ -107,16 +107,38 @@ local mathGrammar = function (_ENV)
          return pl.utils.unpack(t)
          end
 
+   -- TeX uses the regular asterisk (* = U+002A) in superscripts or subscript:
+   -- The TeXbook exercice 18.32 (p. 179, 330) for instance.
+   -- Fonts usually have the asterisk raised too high, so using the Unicode
+   -- asterisk operator U+2217 looks better (= \ast in TeX).
+   local astop = P"*" / luautf8.char(0x2217)
+   -- TeX interprets apostrophes as primes in math mode:
+   -- The TeXbook p. 130 expands ' to ^\prime commands and repeats the \prime
+   -- for multiple apostrophes.
+   -- The TeXbook p. 134: "Then there is the character ', which we know is used
+   -- as an abbreviation for \prime superscripts."
+   -- (So we are really sure superscript primes are really the intended meaning.)
+   -- Here we use the Unicode characters for primes, but the intent is the same.
+   local primes = (
+         P"''''" / luautf8.char(0x2057) + -- quadruple prime
+         P"'''" / luautf8.char(0x2034) + -- triple prime
+         P"''" / luautf8.char(0x2033) + -- double prime
+         P"'" / luautf8.char(0x2032) -- prime
+      ) / function (s)
+            return { id="atom", s }
+         end
+
    START "math"
    math = V"mathlist" * EOF"Unexpected character at end of math code"
    mathlist = (comment + (WS * _) + element)^0
-   supsub = element_no_infix * _ * P"^" * _ * element_no_infix * _ *
-      P"_" * _ * element_no_infix
+   supsub = element_no_infix * _ * primes                      * _ *  P"_" * _ * element_no_infix +
+            element_no_infix * _ * P"^" * _ * element_no_infix * _ *  P"_" * _ * element_no_infix
    subsup = element_no_infix * _ * P"_" * _ * element_no_infix * _ *
       P"^" * _ * element_no_infix
-   sup = element_no_infix * _ * P"^" * _ * element_no_infix
+   sup =  element_no_infix * _ * primes +
```

Comment:
Might not be sufficient for compatibility with TeX: In my 100+ tests, there's (just) one with `\Psi'^2` and we choke on error there. It seems `\Psi'^2 = \Psi^{\prime2}` ($\Psi'^2 = \Psi^{\prime2}$) so we need something more robust in case there's a superscript following the apostrophe-prime...

Reply:
Done.
This Knuth guy is too strong :rofl: -- He doesn't say everything in the text of _The TeXbook,_ but had I looked further, he even had an example just below the explanation, with that construct.